src/test/java/io/redis/examples/StreamExample.java

```diff
@@ -0,0 +1,266 @@
+    //REMOVE_END
+
+    // STEP_START xadd
+    StreamEntryID res1 = jedis.xadd("race:france",new HashMap<String,String>(){{put("rider","Castilla");put("speed","30.2");put("position","1");put("location_id","1");}} , XAddParams.xAddParams());
+
+    System.out.println(res1); // >>> 1701760582225-0
+
+    StreamEntryID res2 = jedis.xadd("race:france",new HashMap<String,String>(){{put("rider","Norem");put("speed","28.8");put("position","3");put("location_id","1");}} , XAddParams.xAddParams());
+
+    System.out.println(res2); // >>> 1701760582225-1
+
+    StreamEntryID res3 = jedis.xadd("race:france",new HashMap<String,String>(){{put("rider","Prickett");put("speed","29.7");put("position","2");put("location_id","1");}} , XAddParams.xAddParams());
+
+    System.out.println(res3); // >>> 1701760582226-0
+
+    //REMOVE_START
+    assertEquals(jedis.xlen("race:france"),3);
+    //REMOVE_END
+
+    //STEP_STAR xrange
+    List<StreamEntry> res4 = jedis.xrange("race:france","1701760582225-0","+",2);
+
+    System.out.println(res4); // >>> [1701760841292-0 {rider=Castilla, speed=30.2, location_id=1, position=1}, 1701760841292-1 {rider=Norem, speed=28.8, location_id=1, position=3}]
+    //STEP_END
+
+    //STEP_START xread_block
+    List<Map.Entry<String, List<StreamEntry>>> res5= jedis.xread(XReadParams.xReadParams().block(300).count(100),new HashMap<String,StreamEntryID>(){{put("race:france",new StreamEntryID());}});
+    System.out.println(
+      res5
+    ); // >>> [race:france=[1701761996660-0 {rider=Castilla, speed=30.2, location_id=1, position=1}, 1701761996661-0 {rider=Norem, speed=28.8, location_id=1, position=3}, 1701761996661-1 {rider=Prickett, speed=29.7, location_id=1, position=2}]]
+    //STEP_END
+
+    //STEP_START xadd_2
+    StreamEntryID res6 = jedis.xadd("race:france",new HashMap<String,String>(){{put("rider","Castilla");put("speed","29.9");put("position","2");put("location_id","1");}} , XAddParams.xAddParams());
+    System.out.println(res6); // >>> 1701762285679-0
+    //STEP_END
+
+    //STEP_START xlen
+    long res7 = jedis.xlen("race:france");
+    System.out.println(res7); // >>> 4
+    //STEP_END
+
+    //STEP_START xadd_id
+    StreamEntryID res8 = jedis.xadd("race:usa", new HashMap<String,String>(){{put("racer","Castilla");}},XAddParams.xAddParams().id("0-1"));
+    System.out.println(res8); // >>> 0-1
+
+    StreamEntryID res9 = jedis.xadd("race:usa", new HashMap<String,String>(){{put("racer","Norem");}},XAddParams.xAddParams().id("0-2"));
+    System.out.println(res9); // >>> 0-2
+    //STEP_END
+
+    //STEP_START xadd_bad_id
+    try {
+      StreamEntryID res10 = jedis.xadd("race:usa", new HashMap<String,String>(){{put("racer","Prickett");}},XAddParams.xAddParams().id("0-1"));
+      System.out.println(res10); // >>> 0-1
+    }
+    catch (JedisDataException e){
+      System.out.println(e); // >>> ERR The ID specified in XADD is equal or smaller than the target stream top item
+    }
+    //STEP_END
+
```

Comment:
I am not founding any example for xadd_7. Can you please suggest where I can found it?

Reply:
Here's the **Redis CLI** example from redis-doc:

```
If you're running Redis 7 or later, you can also provide an explicit ID consisting of the milliseconds part only. In this case, the sequence portion of the ID will be automatically generated. To do this, use the syntax below:

{{< clients-example stream_tutorial xadd_7 >}}
> XADD race:usa 0-* racer Prickett
0-3
{{< /clients-example >}}
```

The Python example is also missing this step. :(

Reply:
There may or may not be a Java for this - as Redis 7 is fairly new, the redis-py library had not implemented this functionality. I am not the most familiar with Jedis but will have @bsbodden let you know whether or not this feature has been implemented yet in Jedis

Reply:
> Please suggest one more thing should I fork 'emb-example' branch for this PR? Because As I can see, 'emb-example' is way behind than master.

@Harsh4902 Yes.